src/results.js

```diff
@@ -31,20 +32,66 @@ class Err {
 	get isErr() { return true }
 	get ok() { return null }
 	get err() { return this._error }
+
 	toString() {
 		if (this._error)
 			return "Error: " + JSON.stringify(this._error)
-		else
-			return "Error"
+		return "Error"
 	}
+
 	toJSON() {
 		let error = (this._error === undefined ? '' : this._error)
 		error = error instanceof Error ? error.toString() : error
 		return { 'Error': error }
 	}
+
+	static _notFound ({ message = 'Not Found', payload, stackTrace } = {}){ 
+		return Err._buildCustomErr('NOT_FOUND', message, payload, stackTrace, 'NotFound', 'notFound')
+	}
+	
+	static _alreadyExists ({ message = 'Entity already exists', payload, stackTrace } = {}) { 
+		return Err._buildCustomErr('ALREADY_EXISTS', message, payload, stackTrace, 'AlreadyExists')
+	}
+	
+	static _invalidEntity ({ message = 'Invalid entity', payload, stackTrace } = {}) { 
+		return Err._buildCustomErr('INVALID_ENTITY', message, payload, stackTrace, 'InvalidEntity')
+	}
+	
+	static _invalidArguments ({ message = 'Invalid arguments', payload, stackTrace } = {}) { 
+		return Err._buildCustomErr('INVALID_ARGUMENTS', message, payload, stackTrace, 'InvalidArguments')
+	}
+	
+	static _permissionDenied ({ message = 'Permission denied', payload, stackTrace } = {}) { 
+		return Err._buildCustomErr('PERMISSION_DENIED', message, payload, stackTrace, 'PermissionDenied')
+	}
+	
+	static _unknown ({ message = 'Unknown Error', payload, stackTrace } = {}) { 
+		return Err._buildCustomErr('UNKNOWN', message, payload, stackTrace, 'Unknown')
+	}
+	
+	static _unimplemented ({ message = 'Unimplemented', payload, stackTrace } = {}) { 
```

Comment:
What's the difference between `UNKNOWN` and `UNIMPLEMENTED`?

Reply:
UNIMPLEMENTED: if you you define an interface and the some implementations not implements all methods, instead you get  "undefined property/functoni", you can implements it and returns UNIMPLEMENTED error code.

UNKNOWN: Is for unhandled error, it can be useful in api-middlewares when you get an js unhandled error, instead you return the nodejs payload error and expose some critical information about your application, you can return a UNKNOWN error

Reply:
Then `UNIMPLEMENTED` should be a exception not a Err. It is a bug, not a alternative path.